src/extent.c

```diff
@@ -541,8 +530,11 @@ extent_recycle_split(tsdn_t *tsdn, pac_t *pac, ehooks_t *ehooks,
 		}
 		if (to_leak != NULL) {
 			extent_deregister_no_gdump_sub(tsdn, pac, to_leak);
+			/* May go down the purge path. */
+			malloc_mutex_unlock(tsdn, &ecache->mtx);
```

Comment:
Is this comment obvious enough for future understanding? Why does this imply we should unlock and relock around this operation?

Reply:
Has something changed that makes extents_abandon_vm not callable under the lock, or has something changed to make this function hold the lock?